lib/mix/lib/mix/tasks/test.ex

```diff
@@ -175,6 +175,24 @@ defmodule Mix.Tasks.Test do
   def run(args) do
     {opts, files} = OptionParser.parse!(args, strict: @switches)
 
+    if opts[:listen_on_stdin] do
+      try do
+        Process.flag(:trap_exit, true)
```

Comment:
Maybe you can use `System.at_exit fn _ -> ... end` to hook into CLI and wait in there (instead of exiting).